app/assets/javascripts/reportModal.js

```diff
@@ -5,6 +5,10 @@ const openModal = () => {
 const closeModal = () => {
     modal.style.display = "none";
 };
+const checkncloseModal = () => {
+    if(reasonForOther.required === true && reasonForOther.value != "" )
+        modal.style.display = "none";
+};
```

Comment:
Please choose a different name for this method. In addition, please re-use the `closeModal` method in order to keep code modular.

```suggestion
const checkAndCloseModal = () => {
    if(reasonForOther.required === true && reasonForOther.value !== "") {
        closeModal();
    }
};
```